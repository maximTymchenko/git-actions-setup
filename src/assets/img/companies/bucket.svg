<svg xmlns="http://www.w3.org/2000/svg" width="66" height="66" viewBox="0 0 66 66">
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <g>
                    <g transform="translate(-1202 -1042) translate(0 516) translate(1135 314) translate(67 212)">
                        <path d="M0 0H66V66H0z"/>
                        <circle cx="33" cy="33" r="33" fill="#3FACE0" opacity=".09"/>
                        <path fill="#3FACE0" fill-rule="nonzero" d="M21 24.5L50 24.5 45.508 42.5 24.484 42.5z" opacity=".557"/>
                        <path fill="#001845" fill-rule="nonzero" d="M12.221 17.002c-.712.05-1.254.663-1.22 1.38.035.717.632 1.275 1.346 1.257h4.461l.958 4.555 4.047 19.271v5.18c.001.725.586 1.314 1.309 1.315h.678c-.02.168-.04.336-.04.505 0 2.488 2.035 4.535 4.51 4.535 2.473 0 4.502-2.047 4.502-4.535 0-.17-.02-.337-.039-.505H36.2c-.02.168-.038.336-.039.505 0 2.488 2.036 4.535 4.51 4.535 2.474 0 4.51-2.047 4.51-4.535 0-.17-.021-.337-.04-.505h1.628c.47.002.906-.249 1.142-.658.236-.41.236-.914 0-1.323-.236-.41-.671-.66-1.142-.659h-22.33v-2.665h22.01c.588-.002 1.103-.396 1.26-.966l5.246-19.425c.107-.397.025-.822-.224-1.149-.249-.327-.635-.518-1.044-.517H20.112l-1.173-5.596h-6.592c-.042-.003-.084-.003-.126 0zM21 25h29l-4.492 17H24.484L21 25zm8.484 4c-.73.009-1.315.6-1.307 1.32v6.346c-.01.473.24.915.653 1.154.414.24.926.24 1.34 0 .414-.24.664-.681.653-1.154V30.32c.004-.351-.136-.69-.387-.938-.252-.248-.595-.386-.952-.382zm5.993 0c-.728.013-1.309.603-1.3 1.32v6.346c-.01.473.24.915.653 1.154.414.24.926.24 1.34 0 .414-.24.664-.68.653-1.154V30.32c.004-.352-.137-.692-.39-.94-.254-.25-.599-.386-.956-.38zm6.009 0c-.352.003-.687.145-.933.393-.246.249-.381.584-.377.931v6.366c0 .723.593 1.31 1.324 1.31.731 0 1.324-.587 1.324-1.31v-6.366c.004-.352-.135-.691-.387-.94-.252-.25-.595-.388-.951-.384zm-14.72 21.295h3.478c.05.172.155.324.155.506 0 1.066-.828 1.904-1.894 1.904-1.066 0-1.904-.838-1.904-1.904 0-.182.114-.333.165-.506zm11.995 0h3.478c.05.173.165.324.165.506 0 1.066-.838 1.904-1.904 1.904-1.066 0-1.904-.838-1.904-1.904 0-.182.114-.333.165-.506z"/>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
