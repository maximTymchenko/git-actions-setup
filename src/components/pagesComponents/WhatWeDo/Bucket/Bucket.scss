@import '../../../../styles/vars';

.bucket {
    background-color: #fff;
    cursor: pointer;
    outline: 2px solid $secondary;
    box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0.4);
    transition: 0.3s linear all;
    width: 100%;
    max-width: 210px;
    height: 100%;
    max-height: 210px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    @media screen and (max-width: $tablet) {
        max-width: 142px;
    }

    &--mobile {
        @media screen and (max-width: $mobile) {
            max-width: 140px;
            justify-content: center;
            align-items: center;
        }
    }

    &--large {
        width: 100%;
        max-width: 200px;
        max-height: 308px;
        // height: 100%;

        // @media screen and (max-width: $medium__desktop) {
        //     max-height: 330px;
        // }

        @media screen and (max-width: $tablet) {
            max-width: 142px;
            justify-content: center;
            align-items: center;
        }
    }

    &--ml {
        margin-right: -70px;
    }

    &--mr {
        margin-right: -9px;
    }

    &--largest {
        width: 100%;
        max-width: 230px;
        max-height: 420px;
        height: 100%;
    }

    &:hover {
        outline: 3px solid $secondary;
        outline-offset: -2px;
    }

    &:active {
        @include button__press;
        box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0);
    }

    &__background {
        margin-top: 20px;
        margin-left: 65px;
        padding-right: 10px;

        @media screen and (max-width: $tablet) {
            margin-left: 45px;
        }

        @media screen and (max-width: $mobile) {
            margin-top: 10px;
        }

        &--moved {
            margin-left: 0px;
            margin-bottom: 25px;
            margin-top: 35px;
            padding: 10px 60px;
        }

        &--center {
            text-align: center;
            margin: 20px auto;
            position: relative;
            left: 5px;
        }
    }

    &__icon {
        height: 150px;

        &--hide {
            display: none;
        }
    }

    &__label {
        margin: 0 15px 10px 30px;
        font-family: 'OktaNeue-Bold';
        font-size: 15px;
        font-weight: bold;
        letter-spacing: 0.5px;
        color: $secondary;
        display: flex;
        align-items: center;

        @media screen and (max-width: $mobile) {
            margin-top: 30px;
        }

        &--displacement {
            margin-left: 60px;
            margin-bottom: 20px;
        }

        &-icon {
            width: 18px;
            height: 18px;
            margin-left: 10px;
        }
    }
}
