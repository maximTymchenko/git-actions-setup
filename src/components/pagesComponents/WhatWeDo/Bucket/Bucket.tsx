import React, { FC } from 'react';
import './Bucket.scss';

import arrRight from '../../../../assets/img/arr-right.svg';

interface IBucketProps {
    icon?: any;
    size: string;
    displacement: string;
    bucketBG: any;
    label: string;
    hide?: string;
    marginBG?: string;
    bucketFunc: Function;
}

const Bucket: FC = (props: IBucketProps) => {
    return (
        <div
            onClick={props.bucketFunc}
            data-testid="bucket"
            className={`bucket bucket--${props.size}`}
        >
            <img
                src={props.bucketBG}
                alt="Background image"
                className={`bucket__background bucket__background--${props.marginBG}`}
            />
            <img
                src={props.icon}
                alt="Bucket icon"
                className={`bucket__icon bucket__icon--${props.hide}`}
            />
            <span className={`bucket__label bucket__label--${props.displacement}`}>
                {props.label}
                <img src={arrRight} alt="Label icon" className="bucket__label-icon" />
            </span>
        </div>
    );
};

export default Bucket;
