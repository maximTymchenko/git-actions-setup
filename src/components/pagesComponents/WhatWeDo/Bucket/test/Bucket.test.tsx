import React from 'react';
import Bucket from '../Bucket';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import bucket from '../../../../../assets/img/companies/bucket.svg';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Bucket icon={bucket} />);
    expect(getByTestId('bucket')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Bucket icon={bucket} />).toJSON();
    expect(tree).toMatchSnapshot();
});
