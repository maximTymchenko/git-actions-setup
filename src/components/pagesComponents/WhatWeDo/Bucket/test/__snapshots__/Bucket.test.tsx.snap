// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="bucket bucket--undefined"
  data-testid="bucket"
>
  <img
    alt="Background image"
    className="bucket__background bucket__background--undefined"
  />
  <img
    alt="Bucket icon"
    className="bucket__icon bucket__icon--undefined"
    src="test-file-stub"
  />
  <span
    className="bucket__label bucket__label--undefined"
  >
    <img
      alt="Label icon"
      className="bucket__label-icon"
      src="test-file-stub"
    />
  </span>
</div>
`;
